File: py/day01.py
# %%
import numpy as np

# %%
def nb_increases(depths, shift=1):
    return sum(depths[shift:] > depths[:-shift])

# %%
if __name__ == "__main__":
    depths = np.loadtxt(f"../data/day01.txt", dtype=int)

    print("Part 1 —", nb_increases(depths))
    print("Part 2 —", nb_increases(depths, shift=3))

File: rs/day01.rs

fn nb_increases(depths: &Vec<usize>, shift: usize) -> usize {
    depths.iter()
        .zip(depths.iter().skip(shift))
        .filter(|(a, b)| b > a).count()
}

fn main() {
    let depths: Vec<usize> = include_str!("../data/day01.txt")
        .lines()
        .map(|x| x.parse().unwrap()) // filter_map(|x| x.unwrap().parse().ok())
        .collect();
    // let depths: Vec<usize> = vec![1, 2, 3, 7, 5, 8];
    println!("Part 1 — {:?}", nb_increases(&depths, 1));
    println!("Part 2 — {:?}", nb_increases(&depths, 3));
}
